arangod/Aql/QueryCache.cpp

```diff
@@ -800,7 +800,7 @@ void QueryCache::excludeSystem() {
 /// @brief determine which lock to use for the cache entries
 unsigned int QueryCache::getPart(TRI_vocbase_t const* vocbase) const {
   uint64_t v = uintptr_t(vocbase);
-  return static_cast<int>(fasthash64_uint64(v, 0xf12345678abcdef) % numberOfParts);
+  return static_cast<int>(fasthash64_uint64(v, 0xf12345678abcdef) & (numberOfParts - 1));
```

Comment:
```suggestion
  return static_cast<int>(fasthash64_uint64(v, 0xf12345678abcdef) % numberOfParts);
```
I would prefer to keep the % version, because it is clearer and it also works correctly should `numberOfParts` be changed to something that is not a power of two. And in case it is a power of two, compilers are smart enough to translate the modulo operation to a bitwise-and themselves.